demo/app/script.ts

```diff
@@ -4,26 +4,29 @@ import { nanoid } from 'nanoid';
 
 import { getRandomName } from './utils/fake-names';
 import { getSpaceNameFromUrl } from './utils/url';
-
 import Spaces from '../../src/Spaces';
 import { renderAvatars, renderSelfAvatar } from './components/avatar-stack';
 import { renderFeatureDisplay } from './components/feature-display';
 
-(async () => {
-  const clientId = nanoid();
-  const ably = new Ably.Realtime.Promise({ authUrl: `/api/ably-token-request?clientId=${clientId}`, clientId });
+const clientId = nanoid();
+
+const ably = new Ably.Realtime.Promise({ authUrl: `/api/ably-token-request?clientId=${clientId}`, clientId });
+
+const spaces = new Spaces(ably);
+
+export const space = spaces.get(getSpaceNameFromUrl(), { offlineTimeout: 60_000 });
+
+export const selfName = getRandomName();
 
-  const spaces = new Spaces(ably);
-  const space = spaces.get(getSpaceNameFromUrl(), { offlineTimeout: 60_000 });
+const memberIsNotSelf = (member) => member.profileData.name !== selfName;
```

Comment:
Should not filter here by connectionId or clientId? these names don't necessarily be unique?